### ISSUE TITLE
Inconsitency in APM trace spans relating LangChain integration

### ISSUE THREAD

**Issue by Peilun-Li** [DataDog/dd-trace-py#11477]
# Environment
- dd-trace-py: 2.16.4
- langchain-core: 0.2.34
- langchain-community: 0.2.1
- langchain-openai: 0.1.8

# Reproduce
A standard langchain AgentExecutor with a retriever tool: https://python.langchain.com/docs/how_to/agent_executor/#retriever and `ainvoke` the agent with a question to trigger the retriever tool.

# Expectation
APM trace span for the request would show three sections:
1. Initial call to LLM (LLM returns desire to usage the retriever tool)
2. Call retriever/vector store
3. Second call to LLM (LLM generates answer with retrieved context)

# Actual behavior
Inconsistent behavior of APM trace span among four test calls (all same output and all triggered tool usage)

First call only shows 1. and 2. above (2. is inferred through request)
![Image](https://github.com/user-attachments/assets/10292fae-71da-4e4a-a5a9-22fc2eec142d)

Second call only shows 1.
![Image](https://github.com/user-attachments/assets/d88c4b8b-6a39-41a3-8f0a-470e1b0110e4)

Third call only shows 1. and 2., with 1. incomplete (only 2 levels compared to 4 levels in the first call)
![Image](https://github.com/user-attachments/assets/c78656ac-937a-4eb1-aa5c-3734d547df9c)

Fourth call only shows 1. and 3., with 3. incomplete (expect the same 4 levels for OpenAI calls but only get 2 levels)
![Image](https://github.com/user-attachments/assets/5611d9e6-133e-48f9-97b3-f2faec53d2cc)

To summarize, it looks like while we do support inject spans for all the expected three sections, they are quite unstable and inconsistent across requests. Could it be due to some setup from our application side, or a ddtrace library issue, or some others? Appreciate any help/insight!
